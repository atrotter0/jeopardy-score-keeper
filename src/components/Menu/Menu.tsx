import React from 'react';

// Styles
import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';

// Icons
import MenuIcon from '@material-ui/icons/Menu';

// Components
import { AppBar, IconButton, Toolbar, Typography } from '@material-ui/core';
import UserIcon from '../UserIcon/UserIcon';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      flexGrow: 1,
    },
    menuButton: {
      marginRight: theme.spacing(2),
    },
    title: {
      flexGrow: 1,
    },
  })
);

const Menu = (): JSX.Element => {
  const classes = useStyles();

  return (
    <AppBar position="fixed">
      <Toolbar>
        <IconButton
          edge="start"
          className={classes.menuButton}
          color="inherit"
          aria-label="menu"
        >
          <MenuIcon />
        </IconButton>
        <Typography variant="h6" className={classes.title}>
          Jeopardy Score Keeper
        </Typography>
        <UserIcon />
      </Toolbar>
    </AppBar>
  );
};

export default Menu;
