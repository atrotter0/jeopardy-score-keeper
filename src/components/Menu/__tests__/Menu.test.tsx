import React from 'react';
import { render } from '@testing-library/react';

// Components
import Menu from '../Menu';

describe('Menu', () => {
  it('should render Menu correctly', () => {
    const { getByText, container } = render(<Menu />);
    const foundText = getByText('Jeopardy Score Keeper');

    expect(foundText).toBeTruthy();
    expect(container).toMatchSnapshot();
  });
});
