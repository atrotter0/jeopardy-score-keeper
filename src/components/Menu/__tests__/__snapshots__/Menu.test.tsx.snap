// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should render Menu correctly 1`] = `
<div>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <button
        aria-label="menu"
        class="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-2 MuiIconButton-colorInherit MuiIconButton-edgeStart"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <h6
        class="MuiTypography-root makeStyles-title-3 MuiTypography-h6"
      >
        Jeopardy Score Keeper
      </h6>
      <button
        aria-controls="menu-appbar"
        aria-label="account of current user"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </header>
</div>
`;
